protos/sources/data-node/api/v2/trading_data.proto

```diff
@@ -924,11 +924,40 @@ service TradingDataService {
 
 // -- Pagination --
 
-// Cursor information for cursor based pagination requests
+// Pagination information for cursor based pagination.
+//
+// All data returned from the API is ordered in some well defined manner. The details of which
+// columns and in which order the sorting is performed depend on which API is being endpoint is 
+// being called, but usually the timestamp of the block in which it data was last updated is the
+// primary sort column.
+//
+// In order to prevent exessively large response messages, and to prevent large queries from
+// starving database resources, the API uses a cursor based pagination mechanism.
+//
+// This pagination message is optionally provided as part of the the request to specify
+// - the starting point within a the total result set to begin the page
+// - size of returned page
+// - the ordering of the data within that page
+//
+// If no pagination message is provided, the API will return the first page of data, using a
+// the default page size. The maximum page size is 1000.
+//
+// To retrieve subsequent pages the caller needs to examine PageInfo structure returned in the
+// response to find a cursor string which uniquely identifies the last row of that page, and
+// then pass that in the 'after' field of the Pagination message in a subsequent request.
+//
+// To paging backwards, take the cursor for the first row of the page from PageInfo and pass that
+// in the 'before' field of the Pagination message.
+//
+// A pagination message that uses first/after as well as last/before is invalid.
 message Pagination {
+    // The number of records to be returned that sort greater than row identified by cursor supplied in 'after'
     optional int32 first = 1;
+    // If paging forwards, the cursor string for the last row of the previous page.
     optional string after = 2;
+    // The number of records to be returned that sort less than row identified by cursor supplied in 'before'
```

Comment:
This line needs to be terminated with a full stop to generate docs appropriately.